### PR TITLE
feat(registration-block): newsletter subscription by default

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Make "Enable newsletter subscription" toggled on by default and improve the UI for when newsletters is not yet properly configured:

| Before | After |
| --- | --- |
| image | image |

There's no need to flood the UI with error messages and settings if the lists are not yet configured.

### How to test the changes in this Pull Request:

1. Make sure you have newsletters installed but no subscription list configured
2. While on master, place a Reader Registration block on a draft page
3. Confirm the "Enable newsletter subscription" is toggled off
4. Toggle it on and confirm the error messages
5. Switch to this branch
6. Start a new draft, place the block and confirm the "Enable newsletter subscription" option is toggled on by default
7. Confirm the new message
8. Enable and configure subscription lists
9. Start a new draft, place the block and confirm you have "Enable newsletter subscription" toggled on with the first list also active by default

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/blocks/reader-registration/edit.js [hunk 1]
```diff
@@ -137,54 +137,62 @@ export default function ReaderRegistrationEdit( {
...
-										setAttributes( { hideSubscriptionInput: ! hideSubscriptionInput } )
-									}
-								/>
-								{ lists.length < 1 && (
-									<div style={ { marginBottom: '1.5rem' } }>
-										<Notice isDismissible={ false } status="error">
-											{ __( 'You must select at least one list.', 'newspack-newsletters' ) }
-										</Notice>
+										{ __(
+											'To enable newsletter subscription, you must enable subscription lists on Newspack Newsletters.',
```
Comment (leogermani) [inline assets/blocks/reader-registration/edit.js:141]: minor: maybe "you must enable at least one Subscription List..."?

**Review by leogermani — APPROVED** [review]
Tests well!

File: assets/blocks/reader-registration/edit.js (hunk 1 above)
Comment (miguelpeixe) [reply assets/blocks/reader-registration/edit.js]: I decided to change a bit more of the phrasing there. WDYT?

image

The use of "configure" in the paragraph should lead to the link.

Comment (leogermani) [reply assets/blocks/reader-registration/edit.js]: Looks good

**Pull request merged**